### ISSUE TITLE
False Positive YODA Condition SIM300

### ISSUE THREAD

**Issue by lukasz-flyrlabs** [astral-sh/ruff#11940]
Keywords searched:
"YODA", "SIM300"

Description:
Ruff incorrectly flags a comparison between a setting from dynaconf (str) and an enum value (str) as a YODA condition. The condition being flagged is not a YODA condition as it involves a configuration setting and an enum, not a plain constant or literal.

Minimal code snippet to reproduce the bug:
```python
from enum import Enum
from dynaconf import Dynaconf

class PriceSource(Enum):
    source1 = "source1"
    source2 = "source2"

settings = Dynaconf(settings_files=["settings.toml"])
# settings.toml content:
# [default]
# PRICE_SOURCE = "source1"

assert settings.PRICE_SOURCE == PriceSource.source1
```

ruff==0.4.9

**Comment by trag1c** [comment]
Corrected code block :)
```py
from enum import Enum
from dynaconf import Dynaconf

class PriceSource(Enum):
    source1 = "source1"
    source2 = "source2"

settings = Dynaconf(settings_files=["settings.toml"])
# settings.toml content:
# [default]
# PRICE_SOURCE = "source1"

assert settings.PRICE_SOURCE == PriceSource.source1
```

**Comment by MichaReiser** [comment]
@AlexWaygood any thoughts on this?

**Comment by AlexWaygood** [comment]
Strictly speaking, I suppose this is a false positive. But to me, it seems like the rule is working as designed, and the heuristic it uses seems pretty reasonable to me. I'm not sure we should change anything here as I haven't seen evidence (yet) that this kind of false positive is particularly common. My advice would be to just `# noqa` it in this instance (or switch off the rule if you find that you have many false positives from this rule on your code).

**Comment by charliermarsh** [comment]
Yeah, I think this is actually reasonable. I'd suggest not using `SCREAMING_SNAKE_CASE` for the setting value if it isn't a constant.
